### PR TITLE
feat: auth expire self / logout

### PR DESCRIPTION
**Pull request by markphelps**
Fixes: FLI-172

- Adds new grpc method `auth.ExpireAuthenticationSelf` that takes optional `expires_at` param to update the current authentication expiration date, defaults to now if not set
- Adds HTTP middleware wrapper to support clearing the `flipt_client_token` and `flipt_client_state` cookies to support logout functionality in the UI

I made it a PUT because that seemed the most appropriate, even though it isn't technically idempotent since continuos calling (without the optional `expiresAt` field) will result in `expiresAt` getting set to NOW() each time). Open to suggestion though

### CODE REVIEW

**Comment by GeorgeMac** [comment]
On the `PUT` vs `GET` this is definitely something to consider. `PUT` could be better in some ways.

I couldn't really make the `callback` a `PUT` or `POST` because the callback is a target that an external provider redirects the browser to.
Using `GET` means you can navigate a browser to the path and get the behaviour.
Is this something we desire for Flipt? As in, the ability to navigate to this expire route and logout.

It is not really an abusable thing given out setup. The `flipt_client_token` cookie is set with `SameSite: strict`.
So, an attacker couldn't force someone to logout. Because they would navigate from another domain and the browser would omit the cookie.
TL;DR `GET` wouldn't be abusable. But it is just the experience we want.

File: internal/server/auth/http.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+func (m Middleware) Handler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPut || r.URL.Path == "/auth/v1/self/expire" {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		for _, cookieName := range []string{stateCookieKey, tokenCookieKey} {
+			cookie := &http.Cookie{
+				Name:   cookieName,
```
Comment (GeorgeMac) [inline internal/server/auth/http.go:31]: Unsure if necessary or not. But should we config this with the same `Domain` from config?

Trying to think of where they might mismatch, and can't think right now where that would be possible. However, if the requested domain doesn't match the domain originaly set in the cookie, this might not clear what we want.
Still though, can't imagine if that is a real edge case or not.

**Review by GeorgeMac — APPROVED** [review]
Nice. I think this looks great.

My feedback is mostly at our discretion to take it or leave it. So I have approved.

File: internal/server/auth/http.go (hunk 1 above)
Comment (markphelps) [reply internal/server/auth/http.go:31]: yah will do
